### PR TITLE
Limited some Maa's mercenary presence

### PR DESCRIPTION
**Pull request by Gaben1025**
## Changelog:

- Added innovation to limit mercenaries coming from dark religions, who have full control over their soldiers

### CODE REVIEW

File: common/men_at_arms_types/_men_at_arms_types.info [hunk 1]
```diff
@@ -36,4 +36,6 @@
 
 	ai_quality = { ... }	# Script value in character scope. Used in addition to a quality calculation (based on TOUGHNESS_SCORE_MULT, etc.) to figure out what the AI should buy/disband
 	fallback_in_hired_troops_if_unlocked = yes # Mercs/holy orders won't have a preference towards this MaA if it is unlocked
+
+	allowed_in_hired_troops = no # Will ban specific MaA from being present in the mercenary pool, the default value is "yes"
```
Comment (zumbak04) [inline common/men_at_arms_types/_men_at_arms_types.info:40]: The Scourge still creates bands. What is the difference between using ghouls and archers?
<details>
<summary>Click to expand</summary>

image

</details>

Comment (Gaben1025) [reply common/men_at_arms_types/_men_at_arms_types.info:40]: I'm not sure if we can outright ban a culture from producing mercs, AFAIK. My goal here was to make sure the special units dont end up in the mercenary band

Comment (zumbak04) [reply common/men_at_arms_types/_men_at_arms_types.info:40]: > My goal here was to make sure the special units dont end up in the mercenary band
It's not a problem.

File: common/religion/doctrines/wc_doctrines_special.txt [hunk 2]
```diff
@@ -212,3 +212,19 @@ doctrine_legions_puppet = {
...
+	
+	special_doctrine_loyal_pawns = {
+		parameters = {
+			is_in_dark_religion_family_trigger = yes
+		}
+		character_modifier = {
+			mercenary_count_mult = -1
+		}
+	}
+}
```
Comment (zumbak04) [inline common/religion/doctrines/wc_doctrines_special.txt:230]: It's not right. You just add the modifier to `doctrine_end_justifies_means`.

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: common/religion/doctrines/wc_doctrines_special.txt (hunk 2 above)
Comment (Gaben1025) [reply common/religion/doctrines/wc_doctrines_special.txt:230]: Oh ok

File: localization/english/religion/wc_core_tenets_l_english.yml.bak [hunk 3]
```diff
@@ -0,0 +1,132 @@
+﻿l_english:
+ tenet_death_syncretism_name:0 "Death Syncretism"
```
Comment (zumbak04) [inline localization/english/religion/wc_core_tenets_l_english.yml.bak:2]: How did this file get there?

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: localization/english/religion/wc_core_tenets_l_english.yml.bak (hunk 3 above)
Comment (zumbak04) [reply localization/english/religion/wc_core_tenets_l_english.yml.bak:2]: Check this PR, looks like you reverted things.

File: common/religion/doctrines/wc_doctrines_special.txt [hunk 4]
```diff
@@ -164,6 +164,10 @@ doctrine_end_justifies_means = {
 			title_creation_cost_mult = -0.5
 			tyranny_gain_mult = -0.5
 			direct_vassal_opinion = -10
+
+			#Makes hyper loyal units not become mercenaries
+			mercenary_count_mult = -1
```
Comment (zumbak04) [inline common/religion/doctrines/wc_doctrines_special.txt:169]: Can't confirm it works. Probably you need to create innovation and use it there.
<details>
<summary>Click to expand</summary>

image

</details>

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: common/religion/doctrines/wc_doctrines_special.txt [hunk 5]
```diff
@@ -163,7 +163,6 @@ doctrine_end_justifies_means = {
 			#tenet_pursuit_of_power
 			title_creation_cost_mult = -0.5
 			tyranny_gain_mult = -0.5
-			direct_vassal_opinion = -10
```
Comment (zumbak04) [inline common/religion/doctrines/wc_doctrines_special.txt:166]: Why you removed it?

**Comment by zumbak04** [comment]
Also, update the changelog.

**Review by zumbak04 — APPROVED** [review]
LGTM!

**Pull request merged**